### PR TITLE
[#44] Add rest server tests

### LINKED ISSUE #44: Add rest server tests
Please add rest server tests to dss-server module.
Check if all the functions works, if not please report the bugs.

You may need to add some http client dependency and complete the test.

Recommended tests:
- basic rest test. (normal json request and its expected response)
- check if contents types and encode module works well

### PR DESCRIPTION
**Pull request by Alencion**
Resolves #44 

add basic rest test
- json post request sends for for comparing expected json with response json.

### CODE REVIEW

**Comment by ztkmkoo** [comment]
@Alencion 
Hi, your branch is out-of-date with develop branch. please check it. Make sure your branch is not conflicted.

File: dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java [hunk 1]
```diff
@@ -95,8 +101,30 @@ public void testSsl() throws Exception {
         assertTrue(dssRestServer.isShutdown());
     }
 
+    @Test
+    public void testJsonPostRequest() throws IOException, InterruptedException {
+        final DssRestServer dssRestServer = new DssRestServer("127.0.0.1", 8181);
+
+        startOnDssJsonRestServer(dssRestServer);
+        await().until(dssRestServer::isActivated);
```
Comment (ztkmkoo) [inline dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java:109]: Hi, 
If there are some bugs, and dssRestServer do not activated. I'm afraid the test will be pending.

Why not let await have timeout or let junit test have timeout property?

Comment (Alencion) [reply dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java:109]: Thank you for your review.

I'll try it.

File: dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java [hunk 2]
```diff
@@ -171,4 +235,28 @@ private static File loadFromTestResources(String path) throws UnsupportedEncodin
...
+        startOnNewThread(() -> {
+            dssRestServer
+                    .addDssRestService(new TestJsonService("test_json", "/test/json", DssRestMethodType.POST));
+            try {
+                dssRestServer.start();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+    }
```
Comment (Alencion) [inline dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java]: I think this method is not general.

It depends on specific conditions.

It does not seem appropriate to write the dssRestSer.addDssRestService method here.

File: dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java [hunk 3]
```diff
@@ -171,4 +235,28 @@ private static File loadFromTestResources(String path) throws UnsupportedEncodin
...
+        final HttpClient httpClient = HttpClientBuilder.create().build();
+        final HttpPost request = new HttpPost(uri);
+
+        final StringEntity params = new StringEntity(sendJsonBody);
+
+        request.addHeader("content-type", "application/json");
+        request.setEntity(params);
+
+        return httpClient.execute(request);
+    }
```
Comment (Alencion) [inline dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java]: this method is too.

It does seem better that parameter add request method type.

Comment (ztkmkoo) [reply dss-server/src/test/java/io/github/ztkmkoo/dss/server/rest/DssRestServerTest.java]: agree

**Comment by ztkmkoo** [comment]
Hi, please update this branch

**Pull request merged**